### PR TITLE
[WTP-1165] Expose helper methods to get text structure

### PR DESCRIPTION
**Pull request by tim-wovn**
`XLIFF::String.source` and `XLIFF::String.target` both throw out the structure of the source and text file.

This structure encodes useful information for html that can be used to disambiguate some situations.

For example, the text `&lt;br&gt;<br>` would be represented with the xliff code: `<source>&lt;br&gt;<ph>&lt;br&gt;</ph></source>`

For that xliff, the corresponding `String.source` is `<br><br>` and it is not possible to construct the original source. This PR exposes the structure via a method that represents the xliff code in this case as a sequence of components like`[['text', 'br'], ['ph', 'br']]` and leaves it to the user to decide the appropriate way to reconstruct the source text.

### CODE REVIEW

File: lib/xliffer/xliff/string.rb [hunk 1]
```diff
@@ -49,7 +61,7 @@ def find_source
         error_message = 'Should have one source tag'
         raise NoElement, error_message unless sources.size == 1
 
-        sources.first.text
+        sources.first
```
Comment (a-bougouin) [inline lib/xliffer/xliff/string.rb:64]: This got me worried, but then I found that it is more consistent with the `find_target` method. Nice!

File: lib/xliffer/xliff/string.rb [hunk 2]
```diff
@@ -41,6 +49,10 @@ def trans_unit?(xml)
         xml.name.downcase == 'trans-unit'
       end
 
+      def text_node_to_components(node)
+        node.children.map { |cell| [cell.name, cell.text] }
```
Comment (a-bougouin) [inline lib/xliffer/xliff/string.rb]: It would be nicer to create our own type with explicit properties.

Comment (tim-wovn) [reply lib/xliffer/xliff/string.rb]: I tried this in 491d16b  What do you think?

Comment (a-bougouin) [reply lib/xliffer/xliff/string.rb]: Looks good, thanks.

**Review by a-bougouin — APPROVED** [review]

File: spec/xliffer/string_spec.rb [hunk 3]
```diff
@@ -79,6 +79,26 @@ module XLIFFer
...
+    describe '#target_components' do
+      it 'Distinguishes un-tagged from tag-labeled sections' do
+        xml = <<-XML
+        <trans-unit id="my id with html">
+          <source>&lt;Hello&gt;<ph>&lt;br&gt;</ph>World</source>
+          <target>&lt;Bonjour&gt;<ph>&lt;br&gt;</ph>le monde</target>
+        </trans-unit>
+        XML
+        trans_unit_node = Nokogiri::XML.parse(xml).xpath('//trans-unit').first
+        puts trans_unit_node
```
Comment (tuantr10) [inline spec/xliffer/string_spec.rb]: Sorry, is this intentional?

File: spec/xliffer/string_spec.rb [hunk 4]
```diff
@@ -98,6 +118,27 @@ module XLIFFer
...
+    describe '#source_components' do
+      it 'Distinguishes un-tagged from tag-labeled sections' do
+        xml = <<-XML
+        <trans-unit id="my id with html">
+          <source>&lt;Hello&gt;<ph>&lt;br&gt;</ph>World</source>
+          <target>&lt;Bonjour&gt;<ph>&lt;br&gt;</ph>le monde</target>
+        </trans-unit>
+        XML
+        trans_unit_node = Nokogiri::XML.parse(xml).xpath('//trans-unit').first
+        puts trans_unit_node
```
Comment (tuantr10) [inline spec/xliffer/string_spec.rb]: And this line too 🙇

File: spec/xliffer/string_spec.rb (hunk 3 above)
Comment (tim-wovn) [reply spec/xliffer/string_spec.rb]: Ah!  Thanks for the catch.  Removed in [68cefdf](https://github.com/WOVNio/xliffer/pull/8/commits/68cefdfb3fd411803d36d3aa30c1bdfe38011332)

File: spec/xliffer/string_spec.rb (hunk 4 above)
Comment (tim-wovn) [reply spec/xliffer/string_spec.rb]: Also removed in [68cefdf](https://github.com/WOVNio/xliffer/pull/8/commits/68cefdfb3fd411803d36d3aa30c1bdfe38011332)  🙇

**Review by tuantr10 — APPROVED** [review]

File: spec/xliffer/string_spec.rb (hunk 4 above)
Comment (tuantr10) [reply spec/xliffer/string_spec.rb]: Thank you!

File: spec/xliffer/string_spec.rb (hunk 3 above)
Comment (tuantr10) [reply spec/xliffer/string_spec.rb]: Thanks!

**Pull request merged**